### PR TITLE
feat: support qrcode embed in markdown and scan in mobile

### PR DESCRIPTION
**Pull request by sheldon66**
## Description
Support
1. inserting variable and qrcode in markdown
2. add qroce scanning module in workbench block

## Key changes
Frontend
change markdown.void component to add variable and qrcode.
change plugin-block-workbench to add qrcode scanning module.

## Showcase

https://github.com/nocobase/nocobase/assets/13678642/67e2171b-4890-4b97-87d3-0695aa201d2e

https://github.com/nocobase/nocobase/assets/13678642/13163c78-b25e-4363-8456-33c3f8ed0281

### CODE REVIEW

File: packages/plugins/@nocobase/plugin-block-workbench/package.json [hunk 1]
```diff
@@ -2,10 +2,12 @@
...
   "version": "1.0.0-alpha.17",
   "main": "dist/server/index.js",
-  "dependencies": {},
   "peerDependencies": {
     "@nocobase/client": "1.x",
     "@nocobase/server": "1.x",
     "@nocobase/test": "1.x"
+  },
+  "dependencies": {
+    "html5-qrcode": "^2.3.8"
```
[the remark this answers is not in the document]
Comment (sheldon66) [reply packages/plugins/@nocobase/plugin-block-workbench/package.json]: 好的，已调整

File: packages/core/client/src/schema-component/antd/markdown/qrcode-webcom.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    } as QRCodeProps;
+    this.root = createRoot(this.shadowRoot as ShadowRoot);
+    this.root.render(<QRCode {...props} />);
+  }
+  disconnectedCallback() {
+    this.root.unmount();
+  }
+}
+let registered = false;
+export function registerQrcodeWebcom() {
```
Comment (zhangzhonghe) [inline packages/core/client/src/schema-component/antd/markdown/qrcode-webcom.tsx:55]:
```suggestion
export function registerQRCodeWebComponent() {
```

File: packages/plugins/@nocobase/plugin-block-workbench/package.json [hunk 3]
```diff
@@ -2,10 +2,13 @@
...
   "main": "dist/server/index.js",
-  "dependencies": {},
   "peerDependencies": {
     "@nocobase/client": "1.x",
     "@nocobase/server": "1.x",
     "@nocobase/test": "1.x"
+  },
+  "devDependencies": {
+    "html5-qrcode": "^2.3.8",
+    "react-router-dom": "^6.23.1"
```
Comment (zhangzhonghe) [inline packages/plugins/@nocobase/plugin-block-workbench/package.json:12]:
```suggestion
    "react-router-dom": "6.x"
```
具体的版本号是在主仓库中确定的：https://github.com/nocobase/nocobase/blob/99e1605bd6affe4abd6702451ce2ebdf5a339c3e/package.json#L50

File: packages/plugins/@nocobase/plugin-block-workbench/src/client/components/qrcode-scanner/index.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      html5QrcodeScanner.clear().catch((error) => {
+        console.error('Failed to clear html5QrcodeScanner. ', error);
+      });
+    };
+  }, [navigate, props]);
+
+  return (
+    <>
+      <div id={qrcodeRegionId} />
+    </>
```
Comment (zhangzhonghe) [inline packages/plugins/@nocobase/plugin-block-workbench/src/client/components/qrcode-scanner/index.tsx:58]:
```suggestion
    <div id={qrcodeRegionId} />
```

**Pull request closed without merging**